### ISSUE TITLE
[BUG] org.opensearch.remotestore.RemoteRestoreSnapshotIT.testRestoreInSameRemoteStoreEnabledIndex is flaky

### ISSUE THREAD

**Issue by ashking94** [opensearch-project/OpenSearch#9116]
**Describe the bug**
`org.opensearch.remotestore.RemoteRestoreSnapshotIT.testRestoreInSameRemoteStoreEnabledIndex` test is flaky. 

**To Reproduce**
./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.RemoteRestoreSnapshotIT.testRestoreInSameRemoteStoreEnabledIndex" -Dtests.seed=CA129BD4A6BD4BD1 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=zh-HK -Dtests.timezone=CNT -Druntime.java=20

**Expected behavior**
Test should pass

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**
Jenkins build failure link - https://build.ci.opensearch.org/job/gradle-check/21872/

**Comment by ashking94** [comment]
failure - https://github.com/opensearch-project/OpenSearch/pull/8758#issuecomment-1665403300

**Comment by ashking94** [comment]
Other instances mentioned in #8279

**Comment by sachinpkale** [comment]
@harishbhakuni Can you please take a look at this bug?

**Comment by peternied** [comment]
Impacted pull request https://github.com/opensearch-project/OpenSearch/pull/9357, see jenkins [testReport](https://build.ci.opensearch.org/job/gradle-check/22785/testReport/org.opensearch.remotestore/RemoteRestoreSnapshotIT/testRestoreInSameRemoteStoreEnabledIndex/)

#### Repro
`./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.RemoteRestoreSnapshotIT.testRestoreInSameRemoteStoreEnabledIndex" -Dtests.seed=DF962FC54758F6F7 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=et-EE -Dtests.timezone=Etc/GMT-4 -Druntime.java=17`

#### Error
```
java.lang.AssertionError: doc with id0 is not present for index testindex1
	at __randomizedtesting.SeedInfo.seed([DF962FC54758F6F7:276070C607309758]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.opensearch.remotestore.RemoteRestoreSnapshotIT.assertDocsPresentInIndex(RemoteRestoreSnapshotIT.java:95)
	at org.opensearch.remotestore.RemoteRestoreSnapshotIT.testRestoreInSameRemoteStoreEnabledIndex(RemoteRestoreSnapshotIT.java:355)
	```

**Comment by harishbhakuni** [comment]
@anasalkouz yes i am looking into this, i was not able to reproduce this locally. ran it for 150+ runs locally and it worked fine. but will fix this.

**Comment by kotwanikunal** [comment]
Just saw this in #9915

**Comment by linuxpi** [comment]
@BhumikaSaini-Amazon can you take a look at this test?

**Comment by harishbhakuni** [comment]
this is an old issue which got fixed in 2.11. ran the test for 100+ runs without any failures.
